### ISSUE TITLE
ModuleNotFoundError: No module named 'setuptools.extern

### ISSUE THREAD

**Issue by GBBBAS** [pypa/setuptools#4498]
### setuptools version

setuptools>=40.8.0

### Python version

3.11.9

### OS

Linux

### Additional environment information

Github CICD Pipeline is failing - the run can be seen here 
https://github.com/rtdip/core/actions/runs/9992039920/job/27655534592

### Description

Get the below error in the output section on `python -m build` when running a CICD Pipeline in Github

### Expected behavior

Python build should complete

### How to Reproduce

python -m build

### Output

```console
Run python -m build
* Creating isolated environment: venv+pip...
* Installing packages in isolated environment:
  - setuptools >= 40.[8](https://github.com/rtdip/core/actions/runs/9992039920/job/27655534592#step:6:9).0
* Getting build dependencies for sdist...
Traceback (most recent call last):
  File "/opt/hostedtoolcache/Python/3.11.[9](https://github.com/rtdip/core/actions/runs/9992039920/job/27655534592#step:6:10)/x64/lib/python3.11/site-packages/pyproject_hooks/_in_process/_in_process.py", line 373, in <module>
    main()
  File "/opt/hostedtoolcache/Python/3.[11](https://github.com/rtdip/core/actions/runs/9992039920/job/27655534592#step:6:12).9/x64/lib/python3.11/site-packages/pyproject_hooks/_in_process/_in_process.py", line 357, in main
    json_out["return_val"] = hook(**hook_input["kwargs"])
                             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/hostedtoolcache/Python/3.11.9/x64/lib/python3.11/site-packages/pyproject_hooks/_in_process/_in_process.py", line 308, in get_requires_for_build_sdist
    return hook(config_settings)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/tmp/build-env-yv5365m2/lib/python3.11/site-packages/setuptools/build_meta.py", line 330, in get_requires_for_build_sdist
    return self._get_build_requires(config_settings, requirements=[])
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/tmp/build-env-yv5365m2/lib/python3.11/site-packages/setuptools/build_meta.py", line 297, in _get_build_requires
    self.run_setup()
  File "/tmp/build-env-yv5365m2/lib/python3.11/site-packages/setuptools/build_meta.py", line 497, in run_setup
    super().run_setup(setup_script=setup_script)
  File "/tmp/build-env-yv5365m2/lib/python3.11/site-packages/setuptools/build_meta.py", line 3[13](https://github.com/rtdip/core/actions/runs/9992039920/job/27655534592#step:6:14), in run_setup
    exec(code, locals())
  File "<string>", line 23, in <module>
ModuleNotFoundError: No module named 'setuptools.extern'
```

**Comment by jaraco** [comment]
To be fair, when I created the `extern` name, I did not make it private as I was unsure if users would need to reach into it. It's possible though unlikely that I even endorsed other users using it, which if I did was a bad choice on my part.

But agreed - the best thing for rtdip to do here is to (a) declare `packaging` in the `build-requires` for the package, and (b) `import packaging` directly. That approach _should_ work for Setuptools before and after v71. Oh, as I look more closely, I see that `packaging` isn't used at all in the setup script. So just remove that line.

Unfortunately, that won't fix previously-published builds. I hope that's an acceptable compromise.
